### ISSUE TITLE
Analytics event: `EXTERNAL_LINK_CLICK`

### ISSUE THREAD

**Issue by dhruvkb** [WordPress/openverse#1078]
## Description

The goal is to 

- add the type for event `EXTERNAL_LINK_CLICK` to the `Events` type.
  https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/types/analytics.ts#L12

- fire this event from the appropriate location in the codebase using the `useAnalytics` composable.
  https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/composables/use-analytics.ts#L11

## Reference

https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/rfcs/analytics/20221006-implementation_plan_frontend.md?plain=1#L287-L305

## Additional information

- [RFC](https://docs.openverse.org/projects/proposals/analytics/20221006-implementation_plan_frontend.html#events)
- [Example usage in `VHomeGallery.vue`](https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/components/VHomeGallery/VHomeGallery.vue#L128-L134)

**Comment by obulat** [comment]
Do we want to send several events when the user clicks on an external link that is also captured by the other events?

For example, we have #1085 for when the user clicks on "Creator" link. But it is also an external link. So, should we send both "EXTERNAL_LINK_CLICK" and "VISIT_CREATOR_LINK" on a click? Or should we have a filter in this issue implementation so that if the user clicks on an external link, we check if it's in one of the special cases (creator, license), and only send `EXTERNAL_LINK_CLICK` if it isn't? @dhruvkb , @zackkrida , @sarayourfriend , @krysal ?

**Comment by dhruvkb** [comment]
In my opinion, `EXTERNAL_LINK_CLICK` is supposed to be like a catch-all fallback in case the click is not being captured by any other category. I see little value in sending more than one event for any interaction.

**Comment by sarayourfriend** [comment]
Agreed that we should not send duplicate events for effectively the same interaction. We'll need to keep in mind that to get the full picture of external navigation we will need to aggregate several events, however.

**Comment by obulat** [comment]
@dhruvkb, it seems that the `component` in the description for this event was taken from another event.

Do we really need the component to be sent with this event?
I implemented this event locally inside the `VLink` component. However, this way the `component` that I get from `useComponent` is `VLink` - which is absolutely useless. If we send the event from the parent of the `VLink` to be able to get the parent component - this might not be a very useful component anyways (I think `url` property might be more useful for this event than the component), and would mean a lot of code duplication all over the app.

What do you think, @WordPress/openverse-frontend ?

**Comment by zackkrida** [comment]
@obulat I think for this event just knowing the page it was made on is sufficient.

**Comment by sarayourfriend** [comment]
If there are multiple instances of this one a single page then an additional disambiguator is probably useful, but only if we care about whether a particular external link location is more useful to users than others. The external source links on the no results page and on a page with results are the same page and it wouldn't be possible to know if the user clicked through the dropdown without some way of disambiguating _which_ instance of that external link they clicked.

I wrote more about this idea [in this PR review which seems not to have been addressed](https://github.com/WordPress/openverse/pull/1151#discussion_r1184430624). @obulat Your review dismissal mentions opening another issue for my feedback. Did that issue get created? It isn't linked from the PR so I wasn't sure what the result was.
